classes/submission/SubmissionDAO.inc.php

```diff
@@ -452,6 +465,114 @@ function getByStatus($status, $stageUserId = null, $reviewUserId, $contextId = n
+		if ($stageId) $params[] = $stageId;
+
+		$result = $this->retrieveRange($sql =
+			'SELECT s.*, ps.date_published,
+				' . $this->_getFetchColumns() . '
+			FROM submissions s
+				LEFT JOIN published_submissions ps ON (s.submission_id = ps.submission_id)
+				LEFT JOIN stage_assignments sa ON (s.submission_id = sa.submission_id)
+				LEFT JOIN submission_files sf ON (s.submission_id = sf.submission_id)
+				LEFT JOIN signoffs so ON (sf.file_id = so.assoc_id) 
+				LEFT JOIN user_groups g ON (g.user_group_id = so.user_group_id)
+				LEFT JOIN review_assignments ra ON (s.submission_id = ra.submission_id)
+				' . ($title?' LEFT JOIN submission_settings ss ON (s.submission_id = ss.submission_id)':'') . '
+				' . ($author?' LEFT JOIN authors au ON (s.submission_id = au.submission_id)':'')
+				. $this->_getFetchJoins() .
+			' WHERE s.date_submitted IS NOT NULL AND ps.date_published IS NULL AND s.user_id <> ? AND s.status <> ?
+				AND (sa.user_id = ? OR (so.user_id = ? AND g.role_id <> ? ) 
+				OR ra.reviewer_id = ?)'
+				. (($contextId)?' AND s.context_id = ?':'')
+				. ($title?' AND (ss.locale = ? AND ss.setting_name = ? AND ss.setting_value LIKE ?)':'') 
+				. ($author?' AND (au.first_name LIKE ? OR au.middle_name LIKE ? OR au.last_name LIKE ?)':'')
+				. ($stageId?' AND s.stage_id = ?':'') .
+			' GROUP BY s.submission_id',
+			$params,
+			$rangeInfo				
+		);
+
+		return new DAOResultFactory($result, $this, '_fromRow');
+	}
+
+	/**
+	 * Get all submissions that are assigned to users other than the passed one.
+	 * @param $userId int
+	 * @param $contextId int optional
+	 * @param $title string|null optional Filter by title.
+	 * @param $author string|null optional Filter by author.
+	 * @param $editor int|null optional Filter by editor name.
+	 * @param $stageId int|null optional Filter by stage id.
+	 * @param $rangeInfo DBResultRange optional
+	 * @return DAOResultFactory
+	 */
+	function getAssignedToOthers($userId, $contextId = null, $title = null, $author = null, $editor = null, $stageId = null, $rangeInfo = null) {
+		$params = $this->_getFetchParameters();
+		$userId = (int) $userId;
+		array_push($params, $userId, STATUS_DECLINED, $userId, ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR);
+		if ($editor) array_push($params, $editorQuery = '%' . $editor . '%', $editorQuery, $editorQuery);
+		if ($contextId) $params[] = $contextId;
+		
+		if ($title) {
+			$params[] = AppLocale::getLocale();
+			$params[] = 'title';
+			$params[] = '%' . $title . '%';
+		}
+		if ($author) array_push($params, $authorQuery = '%' . $author . '%', $authorQuery, $authorQuery);
+		if ($stageId) $params[] = $stageId;
+
+		$result = $this->retrieveRange($sql =
+			'SELECT s.*, ps.date_published,
+				' . $this->_getFetchColumns() . '
+			FROM submissions s
+				LEFT JOIN published_submissions ps ON (s.submission_id = ps.submission_id)
+				LEFT JOIN submission_files sf ON (s.submission_id = sf.submission_id)
```

Comment:
Hmm, that will cause problems with PostgreSQL <9.1.something, as prior to that PostgreSQL required us to actually specify every column. Since then they added smarts to know that grouping by a PRIMARY KEY column was effectively the same thing. I didn't know we were already doing this, but rather than fixing it now, let's keep it consistent.

Just FYI, an alternate approach that would work without the PostgreSQL requirement is to do two joins:
...LEFT JOIN mytable t ON (...)
...LEFT JOIN mytable t2 ON (... AND t2.my_id > t.my_id)
WHERE
t2.my_id IS NULL

The first join brings in the actual content you're looking for, and the second makes sure it's a unique result.